src/common/workflow.utils.ts

```diff
@@ -182,18 +182,20 @@ export const getPendingWorkflowRun = async ({
   octokit: InstanceType<typeof GitHub>;
   logger: log.Logger;
 }): Promise<{ workflowRunId: number; [key: string]: unknown } | undefined> => {
-  const listRunsResult = await octokit.rest.actions.listWorkflowRuns({
-    owner,
-    repo,
-    workflow_id: workflowId,
-    head_sha: commitSha,
-  });
-  logger.debug(
-    `Workflow runs list: ${JSON.stringify(listRunsResult.data, null, 2)}`
-  );
-  const workflowRun = listRunsResult.data.workflow_runs.find((run) =>
-    isPendingStatus(run.status)
+  // The paginate method automatically handles pagination to get all the runs, see more information here:
+  // https://github.com/octokit/octokit.js?tab=readme-ov-file#pagination
+  const workflowRuns = await octokit.paginate(
```

Comment:
Side note / no changes needed: 'octokit.paginate.iterator' + a for loop would be a bit more efficient, but customers are unlikely to have 1000s of workflow runs for a given commit, so this is fine as is.